lib/3scale_toolbox/commands/remote_command/remote_rename.rb

```diff
@@ -17,7 +17,37 @@ def self.command
         end
 
         def run
-          puts "remote rename"
+          validate_input_params
+          begin
+            rename_remote(*arguments[0..1])
+          rescue StandardError => e
+            warn e.message
+            exit 1
+          end
+        end
+
+        def validate_input_params
+          return unless arguments.size != 2
+          puts command.help
+          exit 0
+        end
+
+        def validate_remote_old_name(name)
+          remotes = ThreeScaleToolbox.configuration.remotes
+          raise "Could not rename, old name '#{name}' does not exist." unless remotes.key? name
+        end
+
+        def validate_remote_new_name(name)
+          remotes = ThreeScaleToolbox.configuration.remotes
+          raise "Could not rename, new name '#{name}' already exists." if remotes.key? name
+        end
+
+        def rename_remote(remote_old_name, remote_new_name)
+          validate_remote_old_name remote_old_name
+          validate_remote_new_name remote_new_name
```

Comment:
Why this uses ruby equivalent of global variable? You can't inject the configuration instance and it is loaded from a module.

Reply:
Configuration has setter method and can be injected. Very convenient for testing.

Reply:
It is not injected. It is read from a global variable. Thats quite a difference. It would be injected if it would be part of these messages or object instance.

Reply:
Would be nice to "inject" configuration object as part of options available to all commands. Do not know how, though. Has to be "read" using global variable at some point.

Reply:
The object responsible for instantiating the runner is `cri`. It will parse command line options, instantiate the runner and execute it. That one should be responsible for passing the configuration in the options when instantiating the runner. Then every instance will get its own without the need for global variable. 
Unfortunately it looks like cri does not really support any argument transformation / validation 
https://github.com/3scale/3scale_toolbox/pull/53#discussion_r205044182 like for example clamp has: https://github.com/mdub/clamp#parsing-and-validation-of-options-and-parameters.

Reply:
I mostly agree. 

I am going to open issue for feature request about option validation/transformation as well as arguments.

However, do not going to change CLI framework for now. Let's finish remotes feature and bring some other functionality to have something up and running. Then, CLI framework feature lacking issue can be raised.

Reply:
https://github.com/ddfreyne/cri/issues/67

Reply:
doing some refactor using CLI framework new features